x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -0,0 +1,151 @@
+package keeper
+
+import (
+	"fmt"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/ibc-go/v4/modules/core/exported"
+	ibctmtypes "github.com/cosmos/ibc-go/v4/modules/light-clients/07-tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+func (k Keeper) HandleConsumerMisbehaviour(ctx sdk.Context, misbehaviour ibctmtypes.Misbehaviour) error {
+	if err := k.CheckConsumerMisbehaviour(ctx, misbehaviour); err != nil {
+		return err
+	}
+
+	byzantineValidators, err := k.GetByzantineValidators(ctx, misbehaviour)
+	if err != nil {
+		return err
+	}
+
+	logger := ctx.Logger()
```

Comment:
From my conversation with @sainoe it sounds like it will be very easy to replace this with an actual slash and tombstoning. I think we should do that and then get this merged.